### PR TITLE
Added covariance documentation

### PR DESCRIPTION
**Pull request by Alexandr-Solovev**

### CODE REVIEW

File: cpp/oneapi/dal/algo/covariance/common.hpp [hunk 1]
```diff
@@ -69,11 +71,14 @@ ONEDAL_EXPORT result_option_id get_means_id();
 } // namespace detail
 
 /// Result options are used to define
-/// what should algorithm return
+/// what should an algorithm returns
```
Comment (outoftardis) [inline cpp/oneapi/dal/algo/covariance/common.hpp:74]:
```suggestion
/// what should an algorithm return
```

or

```suggestion
/// what an algorithm returns
```

File: docs/source/api/algorithms/covariance/index.rst [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+.. * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+.. * See the License for the specific language governing permissions and
+.. * limitations under the License.
+.. *******************************************************************************/
+
+=============
+Covariance
+=============
+
+This chapter describes programming interfaces of the covariance algorithms implemented in |short_name|:
```
Comment (outoftardis) [inline docs/source/api/algorithms/covariance/index.rst:21]: probably

```suggestion
This chapter describes programming interfaces of the covariance algorithm implemented in |short_name|:
```

File: docs/source/api/algorithms/covariance/covariance.rst [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+~~~~~~
+.. onedal_class:: oneapi::dal::covariance::compute_result
+
+Operation
+~~~~~~~~~
+.. function:: template <typename Descriptor> \
+              covariance::compute_result compute(const Descriptor& desc, \
+                                      const covariance::compute_input& input)
+
+   :param desc: Covariance algorithm descriptor :expr:`covariance::descriptor`.
```
Comment (outoftardis) [inline docs/source/api/algorithms/covariance/covariance.rst:74]:
```suggestion
   :param desc: Covariance algorithm descriptor :expr:`covariance::descriptor`
```

File: docs/source/onedal/algorithms/covariance/covariance.rst [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+Mathematical formulation
+------------------------
+
+.. _covariance_c_math:
+
+Computing
+---------
+
+Given a set :math:`X` of :math:`n` feature vectors :math:`x_1 = (x_{11}, \ldots, x_{1p}), \ldots, x_n = (x_{n1}, \ldots, x_{np})` of
+dimension :math:`p`, the problem is to compute the sample means or
```
Comment (outoftardis) [inline docs/source/onedal/algorithms/covariance/covariance.rst:36]:
```suggestion
Given a set :math:`X` of :math:`n` :math:`p`-dimensional feature vectors :math:`x_1 = (x_{11}, \ldots, x_{1p}), \ldots, x_n = (x_{n1}, \ldots, x_{np})`,
the problem is to compute the sample means or
```

File: docs/source/onedal/algorithms/covariance/covariance.rst [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+     - :math:`Cov = (v_{ij})`, where :math:`v_{ij}=\frac{1}{n-1}\sum_{k=1}^{n}(x_{ki}-m(i))(x_{kj}-m(j))`, :math:`i=\overline{1,p}`, :math:`j=\overline{1,p}`
+   * - Correlation matrix
+     - :math:`Cor = (c_{ij})`, where :math:`c_{ij}=\frac{v_{ij}}{\sqrt{v_{ii}\cdot v_{jj}}}`, :math:`i=\overline{1,p}`, :math:`j=\overline{1,p}`
+
+.. _covariance_c_math_dense:
+
+Computation method: *dense*
+---------------------------
+The method computes the means or
+variance-covariance matrix or correlation matrix
```
Comment (outoftardis) [inline docs/source/onedal/algorithms/covariance/covariance.rst:58]:
```suggestion
The method computes the means or the
variance-covariance matrix or the correlation matrix
```

File: docs/source/onedal/algorithms/covariance/covariance.rst [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+------------------------
+
+.. _covariance_c_math:
+
+Computing
+---------
+
+Given a set :math:`X` of :math:`n` feature vectors :math:`x_1 = (x_{11}, \ldots, x_{1p}), \ldots, x_n = (x_{n1}, \ldots, x_{np})` of
+dimension :math:`p`, the problem is to compute the sample means or
+covariance matrix or correlation matrix:
```
Comment (outoftardis) [inline docs/source/onedal/algorithms/covariance/covariance.rst:37]:
```suggestion
the covariance matrix or the correlation matrix:
```

**Review by outoftardis — APPROVED** [review]

**Review by makart19 — APPROVED** [review]

**Pull request merged**